src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj

```diff
@@ -176,39 +203,39 @@
    id       :- schema/Int
    config   :- JRubyPuppetConfig
    profiler :- (schema/maybe PuppetProfiler)]
-  (let [{:keys [ruby-load-path gem-home master-conf-dir master-var-dir
+  (let [{:keys [ruby-load-path gem-home
                 http-client-ssl-protocols http-client-cipher-suites]} config]
     (when-not ruby-load-path
-      (throw (Exception.
+      (throw (IllegalStateException.
```

Comment:
Anything is better than an Exception - ISE seemed reasonable, but happy to change to something else